main.py

```diff
@@ -0,0 +1,41 @@
+import streamlit as st
+from transformers import ViltProcessor, ViltForQuestionAnswering
+from PIL import Image
+
+processor = ViltProcessor.from_pretrained("dandelin/vilt-b32-finetuned-vqa")
+model = ViltForQuestionAnswering.from_pretrained("dandelin/vilt-b32-finetuned-vqa")
+
+
+def forward_pass():
+    encoding = processor(image, text, return_tensors="pt")
+    outputs = model(**encoding)
+    logits = outputs.logits
+    return logits.argmax(-1).item()
+
+
+def show_image():
+    image = Image.open(bottom_image)
+    new_image = image.resize((704, 480))
+    st.image(new_image)
+
+
+if __name__ == "__main__":
+    st.title("Visual Question Answering")
```

Comment:
Put this logic under the function. For now, you are working in the global scope and it messes all logic.